parquet-hadoop/src/main/java/org/apache/parquet/filter2/dictionarylevel/DictionaryFilter.java

```diff
@@ -187,10 +196,7 @@ public <T extends Comparable<T>> Boolean visit(NotEq<T> notEq) {
 
     try {
       Set<T> dictSet = expandDictionary(meta);
-      boolean mayContainNull = (meta.getStatistics() == null
-          || !meta.getStatistics().isNumNullsSet()
-          || meta.getStatistics().getNumNulls() > 0);
-      if (dictSet != null && dictSet.size() == 1 && dictSet.contains(value) && !mayContainNull) {
+      if (dictSet != null && dictSet.size() == 1 && dictSet.contains(value)) {
```

Comment:
I just noticed that the `FilterPredicate` does not provide an entry for `IS NULL` or `IS NOT NULL`. This confuses me because `col IS NOT NULL` is not equal to `col != NULL`.

CMIW, `col NOT EQ A` has two meanings as below:
- If A is NULL, it should return an empty list. Because NULL cannot be compared to any value including another NULL.
- Otherwise, it should return a list of values excluding A and NULL.

cc @huaxingao @gszadovszky @shangxinli